src/sign/tests.rs

```diff
@@ -1751,3 +1755,38 @@ fn test_sign1_noncanonical() {
         .verify_signature(aad, |sig, data| verifier.verify(sig, data))
         .is_err());
 }
+
+#[test]
+fn test_sig_structure_data() {
+    let protected = ProtectedHeader {
+        original_data: None,
+        header: Header {
+            alg: Some(Algorithm::Assigned(iana::Algorithm::A128GCM)),
+            key_id: vec![1, 2, 3],
```

Comment:
Maybe use different values for these 2 vecs, to make sure they don't get swapped by the code?

Reply:
Done